### PR TITLE
presence : add default_expires and use in the presence* modules

### LINKED ISSUE #2996: add parameter   presence : default_expires
### Description
Similar to pua:default_expires (see #2994), a default expire value should be added for subscriptions.
The default expiration values are now hard coded for presence modules, in presence_*/add_events.c files.

### Expected behavior
Default expiration value that is configurable.

#### Actual observed behavior
Default expiration value is 3600.

### Possible Solutions
Adding a parameter presence : default_expires and use it in all presence_*/add_events.c files.
I can implement this myself.

### PR DESCRIPTION
**Pull request by adita76**
#### Pre-Submission Checklist

- [x] Commit message has the format required by CONTRIBUTING guide
- [x] Commits are split per component (core, individual modules, libs, utils, ...)
- [x] Each component has a single commit (if not, squash them into one commit)
- [x] No commits to README files for modules (changes must be done to docbook files
in `doc/` subfolder, the README file is autogenerated)

#### Type Of Change
- [ ] Small bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)

#### Checklist:

- [ ] PR should be backported to stable branches
- [x] Tested changes locally
- [x] Related to issue #2996 

#### Description
closes #2996

### CODE REVIEW

File: src/modules/presence_conference/add_events.c [hunk 1]
```diff
@@ -27,14 +27,24 @@
...
 #include <string.h>
 #include "../../core/parser/parse_content.h"
+#include "../../core/sr_module.h"
 #include "../presence/event_list.h"
 #include "presence_conference.h"
 #include "notify_body.h"
 
 int conference_add_events(void)
-{
+{
```
Comment (lnicola) [inline src/modules/presence_conference/add_events.c:36]:
```suggestion
{
```

File: src/modules/presence_conference/add_events.c [hunk 2]
```diff
@@ -43,7 +53,7 @@ int conference_add_events(void)
     event.content_type.s = "application/conference-info+xml";
     event.content_type.len = 31;
 
-    event.default_expires= 3600;
+    event.default_expires= pres_default_expires;
```
Comment (lnicola) [inline src/modules/presence_conference/add_events.c:56]:
```suggestion
    event.default_expires = pres_default_expires;
```

File: src/modules/presence_dialoginfo/add_events.c [hunk 3]
```diff
@@ -43,7 +53,7 @@ int dlginfo_add_events(void)
 	event.content_type.s = "application/dialog-info+xml";
 	event.content_type.len = 27;
 
-	event.default_expires= 3600;
+	event.default_expires= pres_default_expires;
```
Comment (lnicola) [inline src/modules/presence_dialoginfo/add_events.c:56]:
```suggestion
	event.default_expires = pres_default_expires;
```

File: src/modules/presence_mwi/add_events.c [hunk 4]
```diff
@@ -124,7 +134,7 @@ int mwi_add_events(void)
 	event.content_type.s = "application/simple-message-summary";
 	event.content_type.len = 34;
 
-	event.default_expires= 3600;
+	event.default_expires= pres_default_expires;
```
Comment (lnicola) [inline src/modules/presence_mwi/add_events.c:137]:
```suggestion
	event.default_expires = pres_default_expires;
```

File: src/modules/presence_reginfo/add_events.c [hunk 5]
```diff
@@ -25,21 +25,31 @@
...
+
 	/* constructing message-summary event */
 	memset(&event, 0, sizeof(pres_ev_t));
 	event.name.s = "reg";
 	event.name.len = 3;
 
 	event.content_type.s = "application/reginfo+xml";
 	event.content_type.len = 23;
-	event.default_expires= 3600;
+	event.default_expires= pres_default_expires;
```
Comment (lnicola) [inline src/modules/presence_reginfo/add_events.c:52]:
```suggestion
	event.default_expires = pres_default_expires;
```

File: src/modules/presence_xml/add_events.c [hunk 6]
```diff
@@ -49,6 +51,15 @@ int xml_add_events(void)
 {
 	pres_ev_t event;
 
+	/* read default expires from presence, if exists */
+	int pres_default_expires = 3600;
+	modparam_t type = 0;
+	int *param = find_param_export(find_module_by_name("presence"), "default_expires", INT_PARAM, &type);
+	if ( param && (type == INT_PARAM) )
```
Comment (lnicola) [inline src/modules/presence_xml/add_events.c:58]:
```suggestion
	if(param && type == INT_PARAM) 
```

File: src/modules/presence_reginfo/add_events.c [hunk 7]
```diff
@@ -25,21 +25,31 @@
...
 
 int reginfo_add_events(void)
 {
 	pres_ev_t event;
 
+	/* read default expires from presence, if exists */
+	int pres_default_expires = 3600;
+	modparam_t type = 0;
+	int *param = find_param_export(find_module_by_name("presence"), "default_expires", INT_PARAM, &type);
+	if ( param && (type == INT_PARAM) )
```
Comment (lnicola) [inline src/modules/presence_reginfo/add_events.c:40]:
```suggestion
	if(param && type == INT_PARAM) 
```

File: src/modules/presence_mwi/add_events.c [hunk 8]
```diff
@@ -116,6 +117,15 @@ int mwi_add_events(void)
 {
 	pres_ev_t event;
 
+	/* read default expires from presence, if exists */
+	int pres_default_expires = 3600;
+	modparam_t type = 0;
+	int *param = find_param_export(find_module_by_name("presence"), "default_expires", INT_PARAM, &type);
+	if ( param && (type == INT_PARAM) )
```
Comment (lnicola) [inline src/modules/presence_mwi/add_events.c:124]:
```suggestion
	if(param && type == INT_PARAM) 
```

File: src/modules/presence_dialoginfo/add_events.c [hunk 9]
```diff
@@ -35,6 +36,15 @@ int dlginfo_add_events(void)
 {
 	pres_ev_t event;
 
+	/* read default expires from presence, if exists */
+	int pres_default_expires = 3600;
+	modparam_t type = 0;
+	int *param = find_param_export(find_module_by_name("presence"), "default_expires", INT_PARAM, &type);
+	if ( param && (type == INT_PARAM) )
```
Comment (lnicola) [inline src/modules/presence_dialoginfo/add_events.c:43]:
```suggestion
	if(param && type == INT_PARAM) 
```

File: src/modules/presence_conference/add_events.c [hunk 10]
```diff
@@ -27,14 +27,24 @@
...
 int conference_add_events(void)
-{
+{    
     pres_ev_t event;
 
+    /* read default expires from presence, if exists */
+    int pres_default_expires = 3600;
+    modparam_t type = 0;
+    int *param = find_param_export(find_module_by_name("presence"), "default_expires", INT_PARAM, &type);
+    if ( param && (type == INT_PARAM) )
```
Comment (lnicola) [inline src/modules/presence_conference/add_events.c:43]:
```suggestion
	if(param && type == INT_PARAM) 
```

**Comment by miconda** [comment]
@adita76: every module (that needs it) has to get its own parameter, not to lookup the parameter from another module. That's not how usual is done and each module can do some internal optimization to its parameters, so in long term might create hidden issues.

@lnicola: probably the modules' code just need to be run through clang-format.

**Comment by adita76** [comment]
Being a parameter with very simple functionality, I thought it would be easier to use it once than having to configure one for each module.
I'll create one per module, if that's the usual way.

**Comment by miconda** [comment]
Thanks! Modules are for different purposes and each event can be with its own expire settings.

**Pull request merged**